### ISSUE TITLE
How Modifying Self-Hosting Script for Multiple Services on Same Host"

### ISSUE THREAD

**Issue by abulazif** [netbirdio/netbird#1786]
I've been utilizing your self-hosting script, which currently serves on port 443. My aim is to run another WordPress instance alongside the existing service on the same host. Despite attempting various methods, I haven't succeeded in making the necessary changes. Specifically, I need guidance on changing the port of the Netbird service from 443 to another available port to facilitate this setup. My setup is simple, suitable for a small environment. Any assistance on how to accomplish this adjustment effectively would be greatly appreciated. Additionally, here's the command I've been using to deploy Netbird: export NETBIRD_DOMAIN=netbird.example.com; curl -fsSL https://github.com/netbirdio/netbird/releases/latest/download/getting-started-with-zitadel.sh | bash. Thank you for your support.

**Comment by rqi14** [comment]
You can use reverse proxy. Check linuxserver/swag. Thet have a working config

**Comment by abulazif** [comment]
WARN[0000] /root/nbird03/docker-compose.yml: version is obsolete.

My comments:

The script is working great, but there is an issue with the docker-compose.yml file: version is obsolete. I've double-checked my Docker version, and it seems up-to-date. How can I resolve this?

I initially attempted to run the script using a self-signed certificate, which led to several failures.

I successfully generated a Let's Encrypt certificate for my domain using the OpenLiteSpeed Docker version. However, I'm unsure how to automate the renewal process and which tool to use for certificate creation.

Could anybody suggest a tool for managing SSL certificates easily without relying on a web server?

I've tested Zitadel with PostgreSQL manually. Can someone provide guidance on where I need to make adjustments in the script, particularly regarding ports?

Thank you all very much. Happy coding.

**Comment by rqi14** [comment]
Why making it difficult for yourself? Just look at linuxserver/swag. It handles nginx configs and let's encrypt certificate renewal for you. You dont even need to configure anything except enabling the config for your netbird and setting some parameters like your domain name

**Comment by abulazif** [comment]
> Why making it difficult for yourself? Just look at linuxserver/swag. It handles nginx configs and let's encrypt certificate renewal for you. You dont even need to configure anything except enabling the config for your netbird and setting some parameters like your domain name

Did you check how Netbird is installed using a bash script? Can you explain how to use SWAG with Netbird? As per my understanding, it is difficult with a bash script under SWAG. I also do not want to use an API for DNS. I neither want to reserve port 443 for SWAG nor want to use third-party DNS. There are better options available. Insha'Allah.

**Comment by rqi14** [comment]
> > Why making it difficult for yourself? Just look at linuxserver/swag. It handles nginx configs and let's encrypt certificate renewal for you. You dont even need to configure anything except enabling the config for your netbird and setting some parameters like your domain name
> 
> Did you check how Netbird is installed using a bash script? Can you explain how to use SWAG with Netbird? As per my understanding, it is difficult with a bash script under SWAG. I also do not want to use an API for DNS. I neither want to reserve port 443 for SWAG nor want to use third-party DNS. There are better options available. Insha'Allah.

I don't think you will have to reserve 443 for swag. I think you can set it to use DNS challenge mode. If you don't want to use container for your netbitd instance, I think you can point the url to your host in swag. But in that case, it is true that you will still need to solve the changing port issue. I am mentioning the swag thing because you asked how to get pem certificate and automate the renewal process. Swag gives you pem version of certificate, and automates the renewal for you.

**Comment by yabloky** [comment]
> > ПРЕДУПРЕДИТЬ[0000] /root/nbird03/docker-compose.yml: версия устарела.

> > Я успешно сгенерировал сертификат Let's Encrypt для своего домена, используя версию OpenLiteSpeed Docker. 
> 
> удалить первую строку docker-compose.yml. Параметр "version" устарел.

Were you able to run netbird self-host along with openlitespeed reverse proxy? I didn't work, I get a black screen at domen.com/peers.
and an endless loader.
I guess it's because openlitespeed doesn't know how to work with gRPC.
